packages/outputarea/src/widget.ts

```diff
@@ -412,15 +439,76 @@ export class OutputArea extends Widget {
   /**
    * Render and insert a single output into the layout.
    */
-  private _insertOutput(index: number, model: IOutputModel): void {
+  private _insertOutput(
+    index: number,
+    model: IOutputModel,
+    force?: boolean
+  ): void {
+    if (index === 0) {
+      this.trimmedOutputModels = new Array<IOutputModel>();
+    }
+    if (index === this.maxNumberOutputs && !force) {
+      const separatorModel = this.model.contentFactory.createOutputModel({
+        value: {
+          output_type: 'display_data',
+          data: {
```

Comment:
Maybe add a `TODO:` so we can remember to improve this styles in the future.

Reply:
Done. Thx.